code/mob/living/critter/zombie.dm

```diff
@@ -0,0 +1,243 @@
+// Maybe replace with NPC humans later? kill obj for now though
+
+#define NO_EAT 0 // Does not eat brains
+#define EATS_BRAINS 1 // Eats brains and creates new CRITTER zombies
+#define HUMAN_INFECTION 2 // Eats brains and creates new HUMAN zombies
+
+/mob/living/critter/zombie
+	name = "zombie"
+	real_name = "zombie"
+	desc = "BraaAAAinnsSSs..."
+	icon_state = "zombie"
+	can_throw = TRUE
+	can_grab = TRUE
+	can_disarm = TRUE
+	hand_count = 2
+	health_brute = 15
+	health_brute_vuln = 1
+	health_burn = 15
+	health_burn_vuln = 0.1
+	is_npc = TRUE
+	ai_type = /datum/aiHolder/aggressive
+	ai_retaliates = TRUE
+	ai_retaliate_patience = 0
+	ai_retaliate_persistence = RETALIATE_UNTIL_INCAP
+	ai_attacks_per_ability = 5
+
+	var/infection_type = EATS_BRAINS
+
+	New()
+		..()
+		src.add_stam_mod_max("zombie", 100)
+		APPLY_MOVEMENT_MODIFIER(src, /datum/movement_modifier/zombie, src)
+		APPLY_ATOM_PROPERTY(src, PROP_MOB_STAMINA_REGEN_BONUS, "zombie", -5)
+		switch (src.infection_type)
+			if (NO_EAT)
+				return
+			if (EATS_BRAINS)
+				src.abilityHolder.addAbility(/datum/targetable/critter/zombify)
+				return
+			if (HUMAN_INFECTION)
+				src.abilityHolder.addAbility(/datum/targetable/zombie/infect)
+				return
+
+	setup_hands()
+		..()
+		var/datum/handHolder/HH = hands[1]
+		HH.icon = 'icons/mob/hud_human.dmi'
+		HH.limb = new /datum/limb/zombie
+		HH.icon_state = "handl"
+		HH.limb_name = "left zombie arm"
+
+		HH = hands[2]
+		HH.icon = 'icons/mob/hud_human.dmi'
+		HH.limb = new /datum/limb/zombie
+		HH.name = "right hand"
+		HH.suffix = "-R"
+		HH.icon_state = "handr"
+		HH.limb_name = "right zombie arm"
+
+	setup_healths()
+		add_hh_flesh(src.health_brute, src.health_brute_vuln)
+		add_hh_flesh_burn(src.health_burn, src.health_brute_vuln)
+
+	valid_target(mob/living/C)
+		if (istype(C, /mob/living/critter/zombie)) return FALSE
+		if (iszombie(C)) return FALSE
+		if (ishuman(C))
+			var/mob/living/carbon/human/H = C
+			if (istype(H.head, /obj/item/clothing/head/void_crown)) return FALSE
+		return ..()
+
+	critter_ability_attack(mob/target)
+		var/datum/targetable/infect
+		switch (src.infection_type)
+			if (NO_EAT)
+				return FALSE
+			if (EATS_BRAINS)
+				infect = src.abilityHolder.getAbility(/datum/targetable/critter/zombify)
+			if (HUMAN_INFECTION)
+				infect = src.abilityHolder.getAbility(/datum/targetable/zombie/infect)
+		if (!infect?.disabled && infect.cooldowncheck())
+			infect.handleCast(target)
+			return TRUE
+
+	death(var/gibbed)
+		if (istype(src, /mob/living/critter/zombie/biosuit)) return //special death
```

Comment:
This typecheck is a bit gross- maybe put the biosuit-ghost-spawn stuff in `gib()` (and just don't call the parent). Since zombies always gib when they die anyways.